pkg/services/ngalert/notifier/channels/discord.go

```diff
@@ -282,21 +270,32 @@ func (d DiscordNotifier) buildRequest(ctx context.Context, url string, body []by
 	if err != nil {
 		return nil, err
 	}
+
 	if _, err := payload.Write(body); err != nil {
 		return nil, err
 	}
+
 	for _, a := range attachments {
-		part, err := w.CreateFormFile("", a.name)
-		if err != nil {
-			return nil, err
-		}
-		if _, err := io.Copy(part, a.reader); err != nil {
-			return nil, err
+		if a.reader != nil { // We have an image to upload.
+			err = func() error {
+				defer func() { _ = a.reader.Close() }()
```

Comment:
Is there a reason why you don't just call `defer a.reader.Close()`

Reply:
The linter gets angry that we're ignoring the error without explicitly saying "I don't care"